src/Rules.Framework/RulesEngine.cs

```diff
@@ -224,6 +250,13 @@ public Task<RuleOperationResult> UpdateRuleAsync(Rule<TContentType, TConditionTy
                 throw new ArgumentNullException(nameof(rule));
             }
 
+            var validationResult = this.ruleValidator.Validate(rule);
+
+            if (!validationResult.IsValid)
+            {
+                return Task.FromResult(RuleOperationResult.Error(validationResult.Errors.Select(ve => ve.ErrorMessage)));
+            }
```

Comment:
I agree with this, but I think it should be done on `UpdateRuleInternalAsync(...)` after the validation if the rule exists.